lp-core-platform/lp-cp-apis/src/main/java/eu/learnpad/gm/rest/certif/ICertificateDefinitionStorage.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package eu.learnpad.gm.rest.certif;
+
+import java.util.Set;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import eu.learnpad.gm.rest.certif.data.CertificateDefinitionData;
+
+/**
+ *
+ * @author Tom Jorquera - Linagora
+ *
+ */
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+@Path("/gamification")
+public interface ICertificateDefinitionStorage {
```

Comment:
Well... that's the point of the proposal actually. It is supposed to be a starting point for the discussion about what will we actually do or not do.

Personally I agree with the fact that these modifications have a lot of impact, and I am not certain we will manage to reach a complete implementation of the proposal (esp. regarding certificates).
Hopefully the fact that these modifications are somewhat incremental (e.g. scores before certificates) will allow to demonstrate the gamification possibilities in the prototype of the platform, even if not all the possible functionalities are actually implemented.